temporalio/worker/workflow_sandbox/_restrictions.py

```diff
@@ -953,19 +953,17 @@ def r_op(obj: Any, other: Any) -> Any:
 
 
 def _is_restrictable(v: Any) -> bool:
-    return v is not None and not isinstance(
-        v,
-        (
-            bool,
-            int,
-            float,
-            complex,
-            str,
-            bytes,
-            bytearray,
-            datetime.date,  # e.g. datetime.datetime
-        ),
-    )
+    do_not_restrict = [bool, int, float, complex, str, bytes, bytearray]
+    if HAVE_PYDANTIC:
+        # The datetime validator in pydantic_core
+        # https://github.com/pydantic/pydantic-core/blob/741961c05847d9e9ee517cd783e24c2b58e5596b/src/input/input_python.rs#L548-L582
+        # does some runtime type inspection that a RestrictedProxy instance
+        # fails. For this reason we do not restrict date/datetime instances when
+        # Pydantic is being used. Other restricted types such as pathlib.Path
+        # and uuid.UUID which are likely to be used in Pydantic model fields
+        # currently pass Pydantic's validation when wrapped by RestrictedProxy.
+        do_not_restrict.append(datetime.date)  # e.g. datetime.datetime
+    return v is not None and not isinstance(v, tuple(do_not_restrict))
```

Comment:
So `datetime.datetime` will be whitelisted for Pydantic v2 users, but not for v1 users. However, for v1 users have to disable sandboxing of `datetime.datetime` entirely (e.g. per the sample), so `_is_restrictable` is irrelevant for them.

Reply:
Works for me. Though I know we don't prematurely optimize, I wonder if there are performance concerns for making list, appending to list, and turning to tuple every time this call is made. We always made the tuple before. Could make this tuple global if you wanted, but I'm ok with this.

Reply:
Good call, I agree. Performance or not that code looked a bit silly. d915d5b33680152e4286ebbc605920ad832d2a48